sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -148,6 +149,8 @@ private[sql] class SharedState(val sparkContext: SparkContext) extends Logging {
 
 object SharedState {
 
+  URL.setURLStreamHandlerFactory(new SparkUrlStreamHandlerFactory())
```

Comment:
In a [prior PR](https://github.com/apache/spark/pull/16324), FsUrlStreamHandlerFactory is set to JVM URL class directly. @gatorsmile raised a concern that `URL.setURLStreamHandlerFactory` can be called only once per JVM, and that is the motivation of this PR. Either one is OK for me; however we've got to choose one.